examples/allocations.rs

```diff
@@ -0,0 +1,143 @@
+use anyhow::Result;
+use scylla::{Session, SessionBuilder};
+use std::env;
+use std::io::Write;
+use std::sync::Arc;
+
+use tokio::sync::Semaphore;
+
+use stats_alloc::{StatsAlloc, INSTRUMENTED_SYSTEM};
+use std::alloc::System;
+
+#[global_allocator]
+static GLOBAL: &StatsAlloc<System> = &INSTRUMENTED_SYSTEM;
+
+fn print_stats(stats: &stats_alloc::Stats, reqs: f64) {
+    println!(
+        "allocs/req:            {:9.2}",
+        stats.allocations as f64 / reqs
+    );
+    println!(
+        "reallocs/req:          {:9.2}",
+        stats.reallocations as f64 / reqs
+    );
+    println!(
+        "frees/req:             {:9.2}",
+        stats.deallocations as f64 / reqs
+    );
+    println!(
+        "bytes allocated/req:   {:9.2}",
+        stats.bytes_allocated as f64 / reqs
+    );
+    println!(
+        "bytes reallocated/req: {:9.2}",
+        stats.bytes_reallocated as f64 / reqs
+    );
+    println!(
+        "bytes freed/req:       {:9.2}",
+        stats.bytes_deallocated as f64 / reqs
+    );
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+    println!("Connecting to {} ...", uri);
+
+    let session: Session = SessionBuilder::new().known_node(uri).build().await?;
+    let session = Arc::new(session);
+
+    session.query("CREATE KEYSPACE IF NOT EXISTS ks WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1}", &[]).await?;
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS ks.alloc_test (a int, b int, c text, primary key (a, b))",
+            &[],
+        )
+        .await?;
+
+    let prepared_inserts = Arc::new(
+        session
+            .prepare("INSERT INTO ks.alloc_test (a, b, c) VALUES (?, ?, 'abc')")
+            .await?,
+    );
+
+    let parallelism = 256;
+    let sem = Arc::new(Semaphore::new(parallelism));
+
+    let reqs = 100_000usize;
+    print!("Sending {} inserts, hold tight ", reqs);
+    for i in 0..reqs {
```

Comment:
This loop creates a new tokio task on each request. I'm 99% sure that each `tokio::task::spawn` call performs an allocation. I know this loop is modelled after `parallel-prepared`, but perhaps we should change it not to create a separate task in each iteration so that we get more accurate results.